src/key_builder.rs

```diff
@@ -119,6 +119,15 @@ impl KeyBuilder {
     pub fn parse_seq_key(key: &str) -> Option<u64> {
         key.strip_prefix('S')?.parse().ok()
     }
+    
+    /// Get the keypath as a concatenated string
+    pub fn keypath_string(&self) -> String {
+        let mut keypath = String::with_capacity(100);
+        for segment in &self.keypath {
+            keypath.push_str(segment);
+        }
+        keypath
+    }
```

Comment:
This doesn't seem to be used anywhere. If it is used somewhere (or planned), I'd change it to:

```suggestion
    /// Get the keypath as a concatenated string
    pub fn keypath_string(&self) -> String {
        self.keypath.into_iter().collect::<String>()
    }
```